src/rust/ensogl/src/display/scene.rs

```diff
@@ -933,6 +933,12 @@ impl SceneData {
             self.context.viewport(0,0,canvas.width as i32, canvas.height as i32);
         });
     }
+
+    pub fn transform_screen_to_scene_coordinates(&self, position:Vector3<f32>) -> Vector3<f32> {
+        let position = position / self.camera().zoom();
+        let position = Vector4::new(position.x, position.y, position.z, 1.0);
+        (self.camera().inversed_view_matrix() * position).xyz()
```

Comment:
Are you 100% sure it works when scene is both zoomed and panned?

Reply:
![Peek 2020-06-19 11-19](https://user-images.githubusercontent.com/1428930/85117547-53ed1c80-b21f-11ea-9349-36c873ea8200.gif)